File: java/src/main/java/com/meteroid/models/CustomPropertyEntityType.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum CustomPropertyEntityType implements ToQueryParam {
    CUSTOMER("CUSTOMER"),
    SUBSCRIPTION("SUBSCRIPTION"),
    INVOICE("INVOICE"),
    CREDIT_NOTE("CREDIT_NOTE");
    private final String value;

    CustomPropertyEntityType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/custom_property_entity_type.py
# this file is @generated
import enum


class CustomPropertyEntityType(str, enum.Enum):
    CUSTOMER = "CUSTOMER"
    SUBSCRIPTION = "SUBSCRIPTION"
    INVOICE = "INVOICE"
    CREDIT_NOTE = "CREDIT_NOTE"

    def __str__(self) -> str:
        return str(self.value)
